File: storeadmin/api/mutations/condition/updateCondition.ts
import { conditionFragment } from '@/api/fragments'
import gql from 'graphql-tag'

export default gql`
  mutation updateCondition($uuid: ID!, $attributes: ConditionAttributes!) {
    updateCondition(input: { uuid: $uuid, attributes: $attributes }) {
      condition {
        ...conditionFragment
      }
      error
    }
  }
  ${conditionFragment}
`

File: server/app/graphql/mutations/condition/update_condition.rb
class Mutations::Condition::UpdateCondition < Mutations::BaseMutation
  null false

  argument :uuid, ID, required: true
  argument :attributes, InputTypes::Condition, required: true

  field :condition, ObjectTypes::Condition, null: true
  field :error, String, null: true

  def resolve(uuid:, attributes:)
    condition = ::Condition.find_by!(uuid: uuid)
    condition.update!(attributes.to_h)

    { condition: condition.reload }
  rescue => e
    Rails.logger.error(e)
    return GraphQL::ExecutionError.new(e.message)
  end
end
